discord/abc.py

```diff
@@ -1585,9 +1585,19 @@ async def send(
             allowed_mentions = allowed_mentions or AllowedMentions().to_dict()
             allowed_mentions["replied_user"] = bool(mention_author)
 
+        _reference = None
         if reference is not None:
             try:
-                reference = reference.to_message_reference_dict()
+                _reference = reference.to_message_reference_dict()
+                from .message import MessageReference
+
+                if not isinstance(reference, MessageReference):
+                    utils.warn_deprecated(
```

Comment:
Doesn't match the `InvalidArgument` error message below, also why?

Reply:
we discussed this when forwarding came out; before you could pass any `Message` into `reference` to automatically reply to it, but since forwarding was added this behaviour is too vague and should be removed in the next version.